### PR TITLE
Move to outerloop - several  azure-pipeline jobs

### PR DESCRIPTION
**Pull request by briansull**
Remove / move to outer loop:

coreclr-ci (Build Linux arm debug)
coreclr-ci (Build Linux arm release)
coreclr-ci (Build Linux arm64 debug)
coreclr-ci (Build Linux x64 debug)
coreclr-ci (Build Linux x64 release)
coreclr-ci (Build Linux_musl arm64 debug)
coreclr-ci (Build Linux_musl arm64 release)
coreclr-ci (Build Linux_musl x64 debug)
coreclr-ci (Build Linux_rhel6 x64 debug)
coreclr-ci (Build OSX x64 debug)
coreclr-ci (Build OSX x64 release)
coreclr-ci (Build Windows_NT arm debug)
coreclr-ci (Build Windows_NT arm64 debug)
coreclr-ci (Build Windows_NT x64 debug)
coreclr-ci (Build Windows_NT x86 release)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -99,6 +99,21 @@ jobs:
     parameters:
       jobTemplate: build-job.yml
       buildConfig: debug
+      ${{ if eq(variables['Build.DefinitionName'], 'coreclr-ci') }}:
+        platforms:
+        - Windows_NT_x64
+        - Windows_NT_x86
+      ${{ if eq(variables['Build.DefinitionName'], 'coreclr-outerloop') }}:
```
Comment (echesakovMSFT) [inline azure-pipelines.yml]: This wouldn't work - see top level `${{ if }}` which restricts to coreclr-ci build definition

You have two options:
1) relax top level if-predicate (i.e. include coreclr-outerloop in the list of allowed build definitions)
2) clone the whole code block and have two separate blocks for debug build-job - one for coreclr-ci and another for coreclr-outerloop

File: azure-pipelines.yml [hunk 2]
```diff
@@ -124,6 +154,21 @@ jobs:
...
       jobTemplate: build-job.yml
       buildConfig: release
+      ${{ if eq(variables['Build.DefinitionName'], 'coreclr-ci') }}:
+        platforms:
+        - Linux_arm64
+        - Linux_musl_x64
+        - Linux_rhel6_x64
+        - Windows_NT_arm
+        - Windows_NT_arm64
+        - Windows_NT_x64
```
Comment (echesakovMSFT) [inline azure-pipelines.yml]: Same here

**Comment by briansull** [comment]
@echesakovMSFT PTAL

File: azure-pipelines.yml [hunk 3]
```diff
@@ -140,10 +169,9 @@ jobs:
         timeoutInMinutes: 120
 
 #
-# Checked test builds
+# Checked test builds - PullRequest
```
Comment (echesakovMSFT) [inline azure-pipelines.yml]: Is this comment change accidental ? Below this line are PullRequest, Manual and Scheduled builds (not only PullRequest)

Comment (briansull) [reply azure-pipelines.yml]: I'll fix it in the upcoming change for checked jobs

Comment (briansull) [reply azure-pipelines.yml]: I will update

**Comment by briansull** [comment]
@dotnet-build-bot Test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-build-bot Test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness

**Pull request merged**